test/blackbox-tests/test-cases/formatting/non-ascii-characters.t

```diff
@@ -0,0 +1,20 @@
+How the non-ASCII characters are handled, this is also related to the issue #9728
+
+  $ dune format-dune-file <<EOF
+  > ("É")
+  > EOF
+  ("\195\137")
+
+  $ dune format-dune-file <<EOF
+  > (run foo %{bin:é})
+  > EOF
+  > (run foo %{bin:é})
+  File "", line 1, characters 15-16:
+  Error: The character '\195' is not allowed inside %{...} forms
+  run: not found
+  [127]
+
+  $ dune format-dune-file <<EOF
+  > (= "É" "\195\137")
+  > EOF
+  (= "\195\137" "\195\137")
```

Comment:
This part is interesting to add, but not just as a formatting check. We want to make sure that this evaluates to `true`. Maybe as `test-cases/enabled_if/non-ascii.t`?

Reply:
Seeing the result of after parsed, the evaluation will be a comparison between string. Is it really necessary to add that evaluation ?

Reply:
yes, this shows that we do the comparison after interpreting escape directives.

Reply:
I get it.

Reply:
Done in https://github.com/ocaml/dune/pull/9877/commits/9d2d404e7b8ba44b4b302edac91b8102cff1506a.